src/common/common.go

```diff
@@ -93,14 +95,28 @@ func GetPodInStatus(k8Client k8s_client.K8SClient, podNamePrefix string, namespa
 // write tar.gz to pipe in a routine
 // upload tar.gz from pipe to assisted service.
 // close read and write pipes
+// if podName is empty, the controller will upload its own logs from the local controller log file
 func UploadPodLogs(kc k8s_client.K8SClient, ic inventory_client.InventoryClient, clusterId string, podName string, namespace string,
 	sinceSeconds int64, log logrus.FieldLogger) error {
 	log.Infof("Uploading logs for %s in %s", podName, namespace)
-	podLogs, err := kc.GetPodLogsAsBuffer(namespace, podName, sinceSeconds)
-	if err != nil {
-		podLogs = &bytes.Buffer{}
-		podLogs.WriteString(errors.Wrapf(err, "Failed to get logs of pod %s", podName).Error())
+	var podLogs *bytes.Buffer
+	var err error
+	if podName != "" {
+		podLogs, err = kc.GetPodLogsAsBuffer(namespace, podName, sinceSeconds)
+		if err != nil {
+			log.WithError(err).Error("Failed to get logs from kube-api, reading from file")
+		}
 	}
+	if podName == "" || err != nil {
+		log.Infof("Reading logs from file")
+		logs, errF := ioutil.ReadFile(ControllerLogFIle)
+		if errF != nil {
+			log.WithError(errF).Warnf("Failed to read %s", ControllerLogFIle)
+			logs = []byte(fmt.Sprintf("Failed to get logs from kube-api and from file. Read file err %e, kube-api err %e", errF, err))
```

Comment:
Maybe let's return instead? Uploading this line can only damage existing logs...

Although it would probably never happen so no big deal

Reply:
Feel free to unhold after reading this

Reply:
To be sure we will get something if file read will not work for any unknown strange reason